### PR TITLE
Add "Remove group and subgroups" option

### PR DESCRIPTION
**Pull request by tobiasdiez**
Issue #1407 is fixed in the process (long names are handled well by default in javafx, see screenshot below).
snipimage

- [x] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

File: src/main/java/org/jabref/gui/groups/GroupSelector.java [hunk 1]
```diff
@@ -407,13 +403,13 @@ private void showPopup(MouseEvent e) {
             if (node.canBeEdited()) {
                 editGroupPopupAction.setEnabled(false);
                 addGroupPopupAction.setEnabled(false);
-                removeGroupAndSubgroupsPopupAction.setEnabled(false);
+                //removeGroupAndSubgroupsPopupAction.setEnabled(false);
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/gui/groups/GroupSelector.java]: Why only commented out?

File: src/main/java/org/jabref/gui/groups/GroupSelector.java [hunk 2]
```diff
@@ -407,13 +403,13 @@ private void showPopup(MouseEvent e) {
...
                 removeGroupKeepSubgroupsPopupAction.setEnabled(false);
             } else {
                 editGroupPopupAction.setEnabled(true);
                 addGroupPopupAction.setEnabled(true);
                 addGroupPopupAction.setNode(node);
-                removeGroupAndSubgroupsPopupAction.setEnabled(true);
+                //removeGroupAndSubgroupsPopupAction.setEnabled(true);
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/gui/groups/GroupSelector.java]: Why only commented out?

File: src/main/java/org/jabref/gui/groups/GroupSelector.java (hunk 1 above)
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/groups/GroupSelector.java]: The new button is always enabled which is not what you want. Thus this old code functions as a reminder for myself to change it.

**Comment by lenhard** [comment]
If I newly create groups in some bib file and delete them using the options, everything works.

However, I just tested this with our large group file and unfortunately it does not work. I right-click a group (for instance "Life" as the first group in the large bib file), select "Remove group and subgroups", click ok in the message box. Then, nothing happens and the groups stay as they are. There are no messages in the log.

Am I missing something? Can you maybe have a try with our huge bib file?

**Comment by lenhard** [comment]
Looks like other people see a group-related performance degradation: https://github.com/JabRef/jabref/issues/2561#issuecomment-283417338 So, this is most likely not specific to this PR.

**Comment by tobiasdiez** [comment]
@lenhard I can't reproduce the problem with the huge bib file. Thus I will merge this PR for now and would ask you to open a new issue with more details of how to reproduce it.

**Pull request merged**